File: w07/d03/Emmanuel_Tucker/hangman.js
function Hangman(word){
     this.progress = function() {
          var guesses = [];
          var word = ""

word.split("wweee").forEach(function(letter) {
     guesses.push("_");

}) ;
// third circle theorty
     return guesses.join("");

      };
      this.guess = function(letter) {
          return
      };
}


var game =  new Hangman("detective");
// #1 what i want to be outputtedno
console.log("_ _ _ _ _ _ _  _");
console.log(game.progress());

// console.log(false);
// console.log(game.guess("a"));

// console.log(true)
// console.log(game.guess("d"));

// console.log("d_ _ _ _ _ _ _ ")
// console.log(" ")

// function Hangman(word) {
//     this.progress = function() {
//         var guesses = [];

//         word.split("").forEach(function(letter) {
//             guesses.push("_");
//         });

//         return guesses.join(" ");
//     };

//     this.guess = function(letter) {

//     };
// }

// var game = new Hangman("detective");

// console.log("_ _ _ _ _ _ _ _ _");
// console.log(game.progress());

// console.log(false);
// console.log(game.guess("a"));

// console.log(true);
// console.log(game.guess("d"));

// console.log("d _ _ _ _ _ _ _ _");
// console.log(game.progress());

// console.log(true);
// console.log(game.guess("t"));

// console.log("d _ t _ _ t _ _ _");
// console.log(game.progress());

File: w10/d01/Joel_Rosenblatt/hangman/lib/hangman.rb
class Hangman

  def initialize(arry, _nums=Random.new)
    word = arry[_nums.rand(arry.length)]
    @word = word.split("")
    @progress = @word.map { "_" }
    @guesses_so_far = []
    @chances = 0
  end


  def progress
    if @chances < 5
      return @progress.join(" ")
    else
      return @word.join("")
    end
    puts "Incorrect number of guesses: #{@guesses_so_far.length}. #{5 - @guesses_so_far.length} guesses to go"
  end

  def guess(guess_letter)
    if @chances > 5
      return "Stahp!"
    elsif @chances == 5
      @chances += 1
      return "Womp womp.."
    else
      if @word.include? guess_letter
          @word.each_with_index do |letter, index|
            if letter === guess_letter
              @progress[index] = letter
              if @progress == @word
                 return "Woot woot!"
                 @chances = 10
              end
            end
          end
        return true
      else
        @chances += 1
        @guesses_so_far << guess_letter
        return false
      end
    end
  end

  def my_guesses
    return @guesses_so_far.join(", ")
  end

  def giveUp
    return "Looser! The word was: #{@word.join("")}"
    @chances = 10
  end

end
